src/python/WMCore/MicroService/Unified/Common.py

```diff
@@ -80,12 +80,39 @@ def dbsInfo(datasets, dbsUrl):
     return datasetBlocks, datasetSizes, datasetTransfers
 
 
-def phedexValidBlocks(datasets, phedexUrl):
+def getPileupDatasetSizes(datasets, phedexUrl):
     """
     Given a list of datasets, find all their blocks with replicas
-    available, i.e., blocks that have valid files to be processed.
+    available, i.e., blocks that have valid files to be processed,
+    and calculate the total dataset size
     :param datasets: list of dataset names
     :param phedexUrl: a string with the PhEDEx URL
+    :return: a dictionary of datasets and their respective sizes
+    """
+    sizeByDset = {}
+    if not datasets:
+        return sizeByDset
+
+    urls = ['%s/blockreplicas?dataset=%s' % (phedexUrl, dset) for dset in datasets]
+    data = multi_getdata(urls, ckey(), cert())
+
+    for row in data:
+        dataset = row['url'].split('=')[-1]
+        rows = json.loads(row['data'])
+        sizeByDset.setdefault(dataset, 0)  # flat dict in the format of blockName: blockSize
+        for item in rows['phedex']['block']:
```

Comment:
this is phedex specific and you if you're planning on parsing Rucio data you may put this part in function which will properly handle both phedex/rucio data.

Reply:
I'm actually planning to make a separate function for the equivalent API in Rucio. It's more lines of code, but it looks cleaner and easier to cleanup things once we have completely migrated to Rucio.